what4/src/What4/Expr/GroundEval.hs

```diff
@@ -84,6 +85,20 @@ type family GroundValue (tp :: BaseType) where
   GroundValue (BaseArrayType idx b) = GroundArray idx b
   GroundValue (BaseStructType ctx)  = Ctx.Assignment GroundValueWrapper ctx
 
+-- | Return a ground representation of a value, if it is ground.
+asGround :: IsExpr e => e tp -> Maybe (GroundValue tp)
+asGround x =
+  case exprType x of
+    BaseBoolRepr       -> asConstantPred x
+    BaseIntegerRepr    -> asInteger x
+    BaseRealRepr       -> asRational x
+    BaseStringRepr _si -> asString x
+    BaseComplexRepr    -> asComplex x
+    BaseBVRepr _w       -> asBV x
+    BaseFloatRepr _fpp  -> asFloat x
+    BaseStructRepr _   -> asStruct x >>= traverseFC (fmap GVW . asGround)
+    BaseArrayRepr _idx _tp -> Nothing
```

Comment:
This function is strikingly similar to [`What4.Interface.asConcrete`](https://github.com/GaloisInc/what4/blob/c4573936f98d1deb0423af0b489ca63b34aaf4c8/what4/src/What4/Interface.hs#L3202-L3218). Is it feasible to define `asConcrete` in terms of `asGround`? If not, it would be worth mentioning `asConcrete` in `asGround`'s Haddocks, similarly to how we do it fro `groundToSym` (which mentions `What4.Interface.concreteToSym`).